imagery/i.atcorr/create_iwave.py

```diff
@@ -255,72 +255,79 @@ def get_max_wavelength(c, rthresh, fi):
             print("   %s (%inm - %inm)" % (bands[b], min_wavelength, max_wavelength))
 
     # writing...
-    outfile = open(os.path.join(folder, sensor + "_cpp_template.txt"), "w")
-    outfile.write("/* Following filter function created using create_iwave.py */\n\n")
-
-    # single band case
-    if len(bands) == 1:
-        outfile.write("void IWave::%s()\n{\n\n" % (sensor.lower()))
-        outfile.write("    /* %s of %s */\n" % (bands[0], sensor))
-        outfile.write("    static const float sr[%i] = {" % (len(filter_f)))
-        filter_text = pretty_print(filter_f)
-        outfile.write(filter_text)
-
-        # calculate wl slot for band start
-        # slots range from 250 to 4000 at 2.5 increments (total 1500)
-        s_start = int((limits[0] * 1000 - 250) / 2.5)
-
-        outfile.write("\n")
-        outfile.write("    ffu.wlinf = %.4ff;\n" % (limits[0]))
-        outfile.write("    ffu.wlsup = %.4ff;\n" % (limits[1]))
-        outfile.write("    int i = 0;\n")
-        outfile.write("    for(i = 0; i < %i; i++)\tffu.s[i] = 0;\n" % (s_start))
-        outfile.write(
-            "    for(i = 0; i < %i; i++)\tffu.s[%i+i] = sr[i];\n"
-            % (len(filter_f), s_start)
-        )
+    outpath = os.path.join(folder, sensor + "_cpp_template.txt")
+    with open(outpath, "w") as outfile:
```

Comment:
Often, rules changing to use pathlib are related, as they go together with a new pattern. But just this one is fine too ;)

I'll need to look on a computer to see side by side the equivalence of the PR. Or someone else checks before me